kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sequences/matchers.kt

```diff
@@ -49,12 +49,19 @@ fun <T> Sequence<T>.shouldHaveElementAt(index: Int, element: T) = this should ha
 fun <T> Sequence<T>.shouldNotHaveElementAt(index: Int, element: T) = this shouldNot haveElementAt(index, element)
 
 fun <T, S : Sequence<T>> haveElementAt(index: Int, element: T) = object : Matcher<S> {
-   override fun test(value: S) =
-      MatcherResult(
-         value.elementAt(index) == element,
-         { "Sequence should contain $element at index $index" },
-         { "Sequence should not contain $element at index $index" }
+   override fun test(value: S): MatcherResult {
+      val elementAtIndex = value.take(index + 1).mapIndexed { index, t -> index to t }.lastOrNull()
```

Comment:
🔵 Why can't this be `value.elementAt(index)`?

You have the index from `index` if you need to use it downstream

Reply:
> elementAt



> 🔵 Why can't this be `value.elementAt(index)`?
> 
> You have the index from `index` if you need to use it downstream

If we go for `elementAt`, we do not know the actual length of the sequence, we only know that it is too short.

```kotlin
class MyTest: StringSpec() {
    init {
        "blows up" {
            val element = sequenceOf(0, 1, 2).elementAt(4)
            println(element)
        }
    }
}

Sequence doesn't contain element at index 4.
java.lang.IndexOutOfBoundsException: Sequence doesn't contain element at index 4.

```

Because we want to say things like "but the sequence is empty" or "but the sequence only has 3 elements", we need to do some extra work. Does it make sense to you?

Reply:
In that case, can't we use a `.size` or `.count()` and still simplify the rest?